### ISSUE TITLE
Unable to re-authenticate

### ISSUE THREAD

**Issue by jirihelmich** [XiaoMi/ha_xiaomi_home#498]
### Describe the Bug / 描述问题

When trying to re-authenticate the loader in the Configure dialog gets infinitely stuck.

### How to Reproduce / 复现步骤

1. Go to http://homeassistant:8123/config/integrations/integration/xiaomi_home
2. Click configure
3. Click Next
4. Loader loads forever

### Expected Behavior / 预期结果

I would be able to re-authenticate.

### Reproduce Time / 问题复现的时间点

2024-12-29 15:36:00

### Home Assistant Logs / 系统日志

```

2024-12-29 15:20:00.188 ERROR (MainThread) [custom_components.xiaomi_home.miot.miot_client] refresh oauth info error (6349291903, de), 'MIoTOauthClient' object has no attribute '_get_token_async', Traceback (most recent call last):
  File "/config/custom_components/xiaomi_home/miot/miot_client.py", line 449, in refresh_oauth_info_async
    valid_auth_info = await self._oauth.refresh_access_token_async(
                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
        refresh_token=auth_info['refresh_token'])
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/xiaomi_home/miot/miot_cloud.py", line 207, in refresh_access_token_async
    return await self._get_token_async(data={
                 ^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'MIoTOauthClient' object has no attribute '_get_token_async'

2024-12-29 15:20:27.996 ERROR (MainThread) [custom_components.xiaomi_home.miot.miot_client] refresh oauth info error (6349291903, de), 'MIoTOauthClient' object has no attribute '_get_token_async', Traceback (most recent call last):
  File "/config/custom_components/xiaomi_home/miot/miot_client.py", line 449, in refresh_oauth_info_async
    valid_auth_info = await self._oauth.refresh_access_token_async(
                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
        refresh_token=auth_info['refresh_token'])
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/xiaomi_home/miot/miot_cloud.py", line 207, in refresh_access_token_async
    return await self._get_token_async(data={
                 ^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'MIoTOauthClient' object has no attribute '_get_token_async'

2024-12-29 15:22:20.349 ERROR (MainThread) [custom_components.xiaomi_home.miot.miot_client] refresh oauth info error (6349291903, de), 'MIoTOauthClient' object has no attribute '_get_token_async', Traceback (most recent call last):
  File "/config/custom_components/xiaomi_home/miot/miot_client.py", line 449, in refresh_oauth_info_async
    valid_auth_info = await self._oauth.refresh_access_token_async(
                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
        refresh_token=auth_info['refresh_token'])
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/xiaomi_home/miot/miot_cloud.py", line 207, in refresh_access_token_async
    return await self._get_token_async(data={
                 ^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'MIoTOauthClient' object has no attribute '_get_token_async'

2024-12-29 15:22:57.542 ERROR (MainThread) [custom_components.xiaomi_home.miot.miot_client] refresh oauth info error (6349291903, de), 'MIoTOauthClient' object has no attribute '_get_token_async', Traceback (most recent call last):
  File "/config/custom_components/xiaomi_home/miot/miot_client.py", line 449, in refresh_oauth_info_async
    valid_auth_info = await self._oauth.refresh_access_token_async(
                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
        refresh_token=auth_info['refresh_token'])
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/xiaomi_home/miot/miot_cloud.py", line 207, in refresh_access_token_async
    return await self._get_token_async(data={
                 ^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'MIoTOauthClient' object has no attribute '_get_token_async'

2024-12-29 15:25:36.383 ERROR (MainThread) [custom_components.xiaomi_home.miot.miot_client] refresh oauth info error (6349291903, de), 'MIoTOauthClient' object has no attribute '_get_token_async', Traceback (most recent call last):
  File "/config/custom_components/xiaomi_home/miot/miot_client.py", line 449, in refresh_oauth_info_async
    valid_auth_info = await self._oauth.refresh_access_token_async(
                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
        refresh_token=auth_info['refresh_token'])
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/xiaomi_home/miot/miot_cloud.py", line 207, in refresh_access_token_async
    return await self._get_token_async(data={
                 ^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'MIoTOauthClient' object has no attribute '_get_token_async'

2024-12-29 15:28:12.967 ERROR (MainThread) [custom_components.xiaomi_home.miot.miot_client] refresh oauth info error (6349291903, de), 'MIoTOauthClient' object has no attribute '_get_token_async', Traceback (most recent call last):
  File "/config/custom_components/xiaomi_home/miot/miot_client.py", line 449, in refresh_oauth_info_async
    valid_auth_info = await self._oauth.refresh_access_token_async(
                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
        refresh_token=auth_info['refresh_token'])
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/xiaomi_home/miot/miot_cloud.py", line 207, in refresh_access_token_async
    return await self._get_token_async(data={
                 ^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'MIoTOauthClient' object has no attribute '_get_token_async'

2024-12-29 15:35:36.845 ERROR (MainThread) [custom_components.xiaomi_home.miot.miot_client] refresh oauth info error (6349291903, de), 'MIoTOauthClient' object has no attribute '_get_token_async', Traceback (most recent call last):
  File "/config/custom_components/xiaomi_home/miot/miot_client.py", line 449, in refresh_oauth_info_async
    valid_auth_info = await self._oauth.refresh_access_token_async(
                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
        refresh_token=auth_info['refresh_token'])
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/xiaomi_home/miot/miot_cloud.py", line 207, in refresh_access_token_async
    return await self._get_token_async(data={
                 ^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'MIoTOauthClient' object has no attribute '_get_token_async'

2024-12-29 15:36:52.454 INFO (MainThread) [custom_components.xiaomi_home.config_flow] options init, 01JFJN903S8R9VM6TNS1H6S527, de6349291903, {'action_debug': False, 'area_name_rule': 'room', 'cloud_server': 'de', 'ctrl_mode': 'auto', 'hide_non_standard_entities': False, 'home_selected': {'855001013354': {'group_id': '9fc3d3575c83a536', 'home_name': 'Raškova', 'room_info': {'855001046993': 'Study', '855001046994': 'Bathroom', '855001046995': 'Living Room', '855001046996': 'Hall', '855001062536': 'Kitchen'}, 'uid': '6349291903'}}, 'integration_language': 'en', 'nick_name': 'Jiří', 'oauth_redirect_url': 'http://homeassistant.local:8123/api/webhook/11397553492860915841', 'storage_path': '/config/.storage/xiaomi_home', 'uid': '6349291903', 'uuid': '6e497a123d28653a04d33be5be3ddc3a', 'virtual_did': '11397553492860915841'}, {}
2024-12-29 15:36:52.454 INFO (MainThread) [custom_components.xiaomi_home.miot.miot_client] instance exist, 01JFJN903S8R9VM6TNS1H6S527
2024-12-29 15:36:52.461 ERROR (MainThread) [custom_components.xiaomi_home.miot.miot_client] refresh oauth info error (6349291903, de), 'MIoTOauthClient' object has no attribute '_get_token_async', Traceback (most recent call last):
  File "/config/custom_components/xiaomi_home/miot/miot_client.py", line 449, in refresh_oauth_info_async
    valid_auth_info = await self._oauth.refresh_access_token_async(
                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
        refresh_token=auth_info['refresh_token'])
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/xiaomi_home/miot/miot_cloud.py", line 207, in refresh_access_token_async
    return await self._get_token_async(data={
                 ^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'MIoTOauthClient' object has no attribute '_get_token_async'

2024-12-29 15:36:53.604 INFO (MainThread) [custom_components.xiaomi_home.config_flow] async_step_oauth, oauth_url: https://account.xiaomi.com/oauth2/authorize?redirect_uri=http%3A%2F%2Fhomeassistant.local%3A8123%2Fapi%2Fwebhook%2F11397553492860915841&client_id=2882303761520251711&response_type=code&state=17572560996681569029&skip_confirm=False
2024-12-29 15:36:53.604 INFO (MainThread) [custom_components.xiaomi_home.config_flow] async_step_oauth, webhook.async_register: 11397553492860915841

```

### Log Timezone / 日志时区

Prague/Czechia

### Home Assistant Core Version / Home Assistant Core 版本

2024.12.5

### Home Assistant Operation System Version / Home Assistant Operation System 版本

14.1

### Xiaomi Home Integration Version / 米家集成版本

v0.1.3

### Additional Context / 其他说明

Aftera I ran the callback link from the log manually in my browser, I was able to click Authorize on Xiaomi website, but the auth did not go through. However, the next attempt of clicking Next was working.

**Comment by topsworld** [comment]
Hello, you can try the pre-release version v0.1.4b1, and we will release the fix version in the near future.

**Comment by topsworld** [comment]
Hello, the problem has been fixed in v0.1.4.
